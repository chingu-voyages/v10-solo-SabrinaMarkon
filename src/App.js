import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch, Link } from 'redux';
import 'bootstrap/dist/css/bootstrap.min.css';

function App() {
  return (
    <Router>
      <div className="container">
        <h1>Hire a Team!</h1>
      </div>    
    </Router>

  );
}

export default App;
