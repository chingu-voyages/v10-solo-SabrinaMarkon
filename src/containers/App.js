import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

// components
import HowItWorks from '../components/HowItWorks';
import Signup from '../components/Signup';
import Login from '../components/Login';
import PostJob from '../components/PostJob';
import Contact from '../components/Contact';
import UserMain from '../components/UserMain';
import Profile from '../components/Profile';
import Teams from '../components/Teams';
import Jobs from '../components/Jobs';
import Money from '../components/Money';
import NotFound from '../components/NotFound';

export default class App extends Component {
  render() {
    return (
      <Router>
        <div className="container">
          <h1>Hire a Team!</h1>
        </div>
        {/* navigation  */}
        <nav className="navbar navbar-expand-lg navbar-light bg-light">
          <div className="collapse nav-collapse">
            <ul className="navbar-nav mr-auto">
              <li className="navbar-item">
                <Link to="/" className="nav-link">Home</Link>
              </li>
              <li className="navbar-item">
                <Link to="/howitworks" className="nav-link">How It Works!</Link>
              </li>
              <li className="navbar-item">
                <Link to="/signup" className="nav-link">Signup</Link>
              </li>
              <li className="navbar-item">
                <Link to="/login" className="nav-link">Login</Link>
              </li>
              <li className="navbar-item">
                <Link to="/postjob" className="nav-link">Post a Job!</Link>
              </li>
              <li className="navbar-item">
                <Link to="/contact" className="nav-link">Contact</Link>
              </li>
            </ul>
          </div>
        </nav>
      
        {/* not logged in  */}
        {/* <Route exact path="/" component={App} /> */}
        <Route path="/howitworks" component={HowItWorks} />
        <Route path="/signup" component={Signup} />
        <Route path="/login" component={Login} />
        <Route path="/postjob" component={PostJob} />
        <Route path="/contact" component={Contact} />
  
        {/* logged in  */}
        {/* main members area after login - maybe reputation here? */}
        <Route path="/main" component={UserMain} />
        {/* profile should include reputation information (maybe) */}
        <Route path="/profile" component={Profile} />
        {/* teams you are a member of, create a new team, invites, invite others */}
        <Route path="/teams" component={Teams} />
        {/* jobs your team is working on. jobs you want to hire a team for. job list to bid on. */}
        <Route path="/jobs/:type" component={Jobs} />
        {/* your earnings, your teams' earnings per team, owed and by which employer. Payments you owe others you have hired  */}
        <Route path="/money/:type" component={Money} />
  
        {/* custom 404  */}
        <Route component={NotFound} />
      </Router>
    );
  }
}

