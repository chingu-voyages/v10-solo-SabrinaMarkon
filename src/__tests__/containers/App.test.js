import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

// components
import App from '../../containers/App';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

describe('Testing App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should render an h1 heading with the correct text', () => {
    expect(wrapper.find('h1').text().toEqual('App'));
  });
  it('it should match the App snapshot', () => {
    // If snapshot exists, compare. If not, create the snapshot.
    expect(wrapper).toMatchSnapshot();
  });
});