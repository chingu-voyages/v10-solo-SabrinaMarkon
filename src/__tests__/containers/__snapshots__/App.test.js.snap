// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component it should match the App snapshot 1`] = `undefined`;
